### PR TITLE
finished nthreads integration test

### PR DESCRIPTION
**Pull request by ryanlpeterman**
- added integration test to demonstrate that n threads can be handled by the webserver
- usage: python nthreads_test.py <num-threads>

### CODE REVIEW

File: nthreads_test.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		server.kill()
+		return False
+
+	except KeyboardInterrupt:
+		server.kill()
+
+	# clean up temp file
+	finally:
+		os.remove(file_name)
+
```
Comment (adiraju13) [inline nthreads_test.py:70]: will this be called no matter what (even on time out, etc.). Could leave the file open, but shouldn't be too big of an issue.

**Review by adiraju13 — COMMENTED** [review]
Looks good overall, feel free to merge

File: nthreads_test.py (hunk 1 above)
Comment (ryanlpeterman) [reply nthreads_test.py:70]: it will be called no matter what, finally is intense

**Comment by ryanlpeterman** [comment]
Hi. if you're grading this, we were told by Shaghayegh, our TA, to include the conversations we had in messenger to count for credit about the code since we were getting lower grades on the code review comments:
selection_080

**Pull request merged**